libraries/standard/mqtt/cbmc/proofs/MQTT_SerializeConnect/MQTT_SerializeConnect_harness.c

```diff
@@ -28,35 +28,43 @@
 
 void harness()
 {
-    MQTTConnectInfo_t * pConnectInfo = NULL;
-    MQTTPublishInfo_t * pWillInfo = NULL;
-    size_t remainingLength = 0;
-    MQTTFixedBuffer_t * pBuffer = NULL;
-    size_t packetSize = 0;
+    MQTTConnectInfo_t * pConnectInfo;
+    MQTTPublishInfo_t * pWillInfo;
+    size_t remainingLength;
+    MQTTFixedBuffer_t * pFixedBuffer;
+    size_t packetSize;
     MQTTStatus_t status = MQTTSuccess;
 
-    pConnectInfo = allocateMqttConnectInfo( pConnectInfo );
+    pConnectInfo = allocateMqttConnectInfo( NULL );
     __CPROVER_assume( isValidMqttConnectInfo( pConnectInfo ) );
 
-    pWillInfo = allocateMqttPublishInfo( pWillInfo );
+    pWillInfo = allocateMqttPublishInfo( NULL );
     __CPROVER_assume( isValidMqttPublishInfo( pWillInfo ) );
 
-    pBuffer = allocateMqttFixedBuffer( pBuffer );
-    __CPROVER_assume( isValidMqttFixedBuffer( pBuffer ) );
+    pFixedBuffer = allocateMqttFixedBuffer( NULL );
+    __CPROVER_assume( isValidMqttFixedBuffer( pFixedBuffer ) );
 
     /* Before calling MQTT_SerializeConnect() it is up to the application to make
-    * sure that the information in MQTTConnectInfo_t and MQTTPublishInfo_t can
-    * fit into the MQTTFixedBuffer_t. It is a violation of the API to call
-    * MQTT_SerializeConnect without first calling MQTT_GetConnectPacketSize(). */
+     * sure that the information in MQTTConnectInfo_t and MQTTPublishInfo_t can
+     * fit into the MQTTFixedBuffer_t. It is a violation of the API to call
+     * MQTT_SerializeConnect() without first calling MQTT_GetConnectPacketSize(). */
     if( pConnectInfo != NULL )
     {
-        status = MQTT_GetConnectPacketSize( pConnectInfo, pWillInfo, &remainingLength, &packetSize );
+        /* packetSize must be non-NULL in order for the verification to proceed.
```

Comment:
packetSize is not NULL or the 4th parameter of MQTT_GetConnectPacketSize is not NULL?

```
The output parameter pPacketSize of the function MQTT_GetConnectPacketSize must not be NULL. The packetSize ...
```

Reply:
The 4th parameter which is packetSize must not be null yes.